test/test_framework.py

```diff
@@ -894,10 +898,37 @@ def test_stored_dict_repr(self):
         self.assertEqual(repr(StoredDict(None, {})), "ops.framework.StoredDict()")
         self.assertEqual(repr(StoredDict(None, {"a": 1})), "ops.framework.StoredDict({'a': 1})")
 
+    def test_stored_dict_equality(self):
+        self.assertEqual(StoredDict(None, {}), {})
+        self.assertEqual(StoredDict(None, {"a": 1}), {"a": 1})
+
+    def test_stored_dict_yaml(self):
+        self.assertEqual(stdlib_yaml.dump(StoredDict(None, {})), '{}\n')
+        self.assertEqual(stdlib_yaml.dump(StoredDict(None, {"a": 1, "b": 2})), 'a: 1\nb: 2\n')
+
+    def test_stored_dict_json(self):
+        self.assertEqual(stdlib_json.dumps(StoredDict(None, {})), '"{}"')
+        self.assertEqual(stdlib_json.dumps(
+            StoredDict(None, {"a": 1, "b": 2})),
+            '"{\\"a\\": 1, \\"b\\": 2}"' # NOQA
+        )
```

Comment:
If we don't monkeypatch, we'd need to add a ```cls=CustomEncoder```-type parameter to each of the dumps calls, or we could add our own ```dumps``` call to our own json.py and hide that detail there.

In my opinion, that's desirable to monkeypatching unless there is a stronger reason to need to do so, in my opinion.

There is one other path, atlhough I don't know if it's a good idea or not, and that is to subclass StoredDict/StoredList off of the base dict and list classes.  The json module's encoding logic uses isinstance checks to check types, and subclasses of the stdlib types would pass those checks.  That being said, I'm not sure that I really like that idea. :-)